scrapyd_k8s/api.py

```diff
@@ -11,6 +12,16 @@
 config = Config()
 repository = (config.repository_cls())(config)
 launcher = (config.launcher_cls())(config)
+scrapyd_config = config.scrapyd()
+
+config_username = scrapyd_config.get("username")
+config_password = scrapyd_config.get("password")
+
+if not (config_username is None or config_password is None):
+    basic_auth = BasicAuth(app)
+    app.config['BASIC_AUTH_USERNAME'] = config_username
+    app.config['BASIC_AUTH_PASSWORD'] = config_password
+    app.config["BASIC_AUTH_FORCE"] = True
```

Comment:
Can you move this to a function at the end of the file? It is not central to this application, and more like an extra detail you can read about after the rest.
And then here the function can be called to setup authentication for the `app`.

Reply:
I also wonder if the function should be called here or in `run()`, any thoughts?